packages/graphql/src/translate/queryAST/ast/filters/property-filters/CypherFilter.ts

```diff
@@ -0,0 +1,117 @@
+import { createComparisonOperation } from "../../../utils/create-comparison-operator";
+import { createPointOperation } from "../../../utils/create-point-operation";
+import type { QueryASTContext } from "../../QueryASTContext";
+import type { QueryASTNode } from "../../QueryASTNode";
+import type { CustomCypherSelection } from "../../selection/CustomCypherSelection";
+import type { FilterOperator } from "../Filter";
+import { Filter } from "../Filter";
+
+/** A property which comparison has already been parsed into a Param */
+export class CypherFilter extends Filter {
+    private returnVariable: Cypher.Variable = new Cypher.Variable();
+    private attribute: AttributeAdapter;
+    private selection: CustomCypherSelection;
+    private operator: FilterOperator;
+    protected comparisonValue: unknown;
+
+    constructor({
+        selection,
+        attribute,
+        operator,
+        comparisonValue,
+    }: {
+        selection: CustomCypherSelection;
+        attribute: AttributeAdapter;
+        operator: FilterOperator;
+        comparisonValue: unknown;
+    }) {
+        super();
+        this.selection = selection;
+        this.attribute = attribute;
+        this.operator = operator;
+        this.comparisonValue = comparisonValue;
+    }
+
+    public getChildren(): QueryASTNode[] {
+        return [this.selection];
+    }
+
+    public print(): string {
+        return `${super.print()} [${this.attribute.name}] <${this.operator}>`;
+    }
+
+    public getSubqueries(context: QueryASTContext): Cypher.Clause[] {
+        const { selection: cypherSubquery, nestedContext } = this.selection.apply(context);
+
+        const clause = Cypher.concat(
+            cypherSubquery,
+            new Cypher.Return([nestedContext.returnVariable, this.returnVariable])
+        );
+
+        return [clause];
+    }
+
+    public getPredicate(_queryASTContext: QueryASTContext): Cypher.Predicate {
+        const operation = this.createBaseOperation({
+            operator: this.operator,
+            property: this.returnVariable,
+            param: new Cypher.Param(this.comparisonValue),
+        });
+
+        return operation;
+    }
+
+    private coalesceValueIfNeeded(expr: Cypher.Expr): Cypher.Expr {
+        if (this.attribute.annotations.coalesce) {
+            const value = this.attribute.annotations.coalesce.value;
+            const literal = new Cypher.Literal(value);
+            return Cypher.coalesce(expr, literal);
+        }
+        return expr;
+    }
+
+    /** Returns the default operation for a given filter */
+    private createBaseOperation({
+        operator,
+        property,
+        param,
+    }: {
+        operator: FilterOperator;
+        property: Cypher.Expr;
+        param: Cypher.Expr;
+    }): Cypher.ComparisonOp {
+        const coalesceProperty = this.coalesceValueIfNeeded(property);
+
+        if (operator === "DISTANCE") {
```

Comment:
From what I see, I believe here there is a bug in the implementation, it uses the `createPointOperation` depending on the operator, while it should use `createPointOperation` depending on the AttributeType IMO. I believe that the same error will be present also for Duration Filters or any other filters that generate the predicate differently from PropertyFilter.